app/services/foreman_ansible/ansible_report_scanner.rb

```diff
@@ -5,19 +5,14 @@ module ForemanAnsible
   # sets the origin of the report to 'Ansible'
   class AnsibleReportScanner
     class << self
-      def scan(report, logs)
-        if (is_ansible = ansible_report?(logs))
-          report.origin = 'Ansible'
-        end
-        is_ansible
+      def add_reporter_data(report, raw); end
+
+      def identify_origin(raw)
+        'Ansible' if ansible_report?(raw)
       end
 
-      def ansible_report?(logs)
-        return false if logs.blank?
-        logs.any? do |log|
-          log['log'].fetch('messages', {}).
-            fetch('message', '') =~ /"_ansible_parsed"/
-        end
+      def ansible_report?(raw)
+        raw['reporter'] == 'ansible'
```

Comment:
Should we keep the backward compatibility?
By renaming the old method to `ansible_27_report?` and
```suggestion
        raw['reporter'] == 'ansible' || ansible_27_report?(raw['logs'] || [])
```

Reply:
And we should then have two fixtures and two tests for it I'd guess :thinking: